Build 1754 for main with status FAILURE - Build&Test / libbeat-goIntegTest / TestTemplateDisabled

## :broken_heart: Tests Failed


<!-- BUILD BADGES-->
> _the below badges are clickable and redirect to their specific view in the CI or DOCS_
[![Pipeline View](https://img.shields.io/badge/pipeline-pipeline%20-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2Fmain/detail/main/1754//pipeline) [![Test View](https://img.shields.io/badge/test-test-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2Fmain/detail/main/1754//tests) [![Changes](https://img.shields.io/badge/changes-changes-green)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2Fmain/detail/main/1754//changes) [![Artifacts](https://img.shields.io/badge/artifacts-artifacts-yellow)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2Fmain/detail/main/1754//artifacts) [![preview](https://img.shields.io/badge/docs-preview-yellowgreen)](http://beats_null.docs-preview.app.elstc.co/diff)  [![preview](https://img.shields.io/badge/elastic-observability-blue)](https://ci-stats.elastic.co/app/apm/services/beats-ci/transactions/view?rangeFrom=2023-07-30T13:47:52.405Z&rangeTo=2023-07-30T14:07:52.405Z&transactionName=BUILD+Beats%2Fbeats%2Fmain&transactionType=job&latencyAggregationType=avg&traceId=835ffdaed1785a063503a591e2d4245a&transactionId=53cddbcb0f3c952c)

<!-- BUILD SUMMARY-->
<details><summary>Expand to view the summary</summary>
<p>

#### Build stats




* Start Time: 2023-07-30T13:57:52.405+0000


* Duration: 75 min 24 sec




#### Test stats :test_tube:

| Test         | Results                         |
| ------------ | :-----------------------------: |
| Failed       | 1  |
| Passed       | 27592  |
| Skipped      | 2008 |
| Total        | 29601   |


</p>
</details>

<!-- TEST RESULTS IF ANY-->

  ### Test errors [![1](https://img.shields.io/badge/1%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2Fmain/detail/main/1754//tests)
<details><summary>Expand to view the tests failures</summary><p>
    
    
##### `Build&Test / libbeat-goIntegTest / TestTemplateDisabled – github.com/elastic/beats/v7/libbeat/tests/integration`


<ul>
<details><summary>Expand to view the error details</summary><p>

```
 Failed 
 ```
</p></details>
<details><summary>Expand to view the stacktrace</summary><p>

```
 === RUN   TestTemplateDisabled
    template_test.go:296: 
        	Error Trace:	/var/lib/jenkins/workspace/main-1754-c7e2a3ab-fdf4-42c7-944a-9225a9ea572c/src/github.com/elastic/beats/libbeat/tests/integration/framework.go:268
        	            				/var/lib/jenkins/workspace/main-1754-c7e2a3ab-fdf4-42c7-944a-9225a9ea572c/src/github.com/elastic/beats/libbeat/tests/integration/template_test.go:296
        	Error:      	Condition never satisfied
        	Test:       	TestTemplateDisabled
    framework.go:352: Temporary directory saved: /var/lib/jenkins/workspace/main-1754-c7e2a3ab-fdf4-42c7-944a-9225a9ea572c/src/github.com/elastic/beats/libbeat/build/integration-tests/TestTemplateDisabled-1690728364
--- FAIL: TestTemplateDisabled (20.32s)
 
 ```
</p></details>
</ul>
    
</p></details>


<!-- STEPS ERRORS IF ANY -->














### Steps errors [![4](https://img.shields.io/badge/4%20-red)](https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2Fmain/detail/main/1754//pipeline)
  <details><summary>Expand to view the steps failures</summary>
  <p>


  
  
  
  
##### `libbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 8 min 37 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/main/runs/1754/steps/13190/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `libbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 4 min 14 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/main/runs/1754/steps/18563/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `libbeat-goIntegTest - mage goIntegTest`
<ul>
<li>Took 4 min 54 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/main/runs/1754/steps/20101/log/?start=0">here</a></li>
<li>Description: <code>mage goIntegTest</code></l1>
</ul>
  
  
  
  
##### `Error signal`
<ul>
<li>Took 0 min 0 sec . View more details <a href="https://beats-ci.elastic.co//blue/rest/organizations/jenkins/pipelines/Beats/pipelines/beats/pipelines/main/runs/1754/steps/21186/log/?start=0">here</a></li>
<li>Description: <code>Error "hudson.AbortException: script returned exit code 1"</code></l1>
</ul>
  
  </p>
  </details>

--- a/libbeat/tests/integration/framework.go
+++ b/libbeat/tests/integration/framework.go
@@ -31,6 +31,7 @@ import (
 	"net/url"
 	"os"
 	"path/filepath"
+	"regexp"
 	"strings"
 	"sync"
 	"testing"
@@ -118,6 +119,37 @@ func NewBeat(t *testing.T, beatName, binary string, args ...string) *BeatProc {
 		stdout:     stdoutFile,
 		stderr:     stderrFile,
 	}
+	t.Cleanup(func() {
+		if !t.Failed() {
+			return
+		}
+		stderr, err := os.ReadFile(filepath.Join(tempDir, "stderr"))
+		if err != nil {
+			t.Logf("error reading stderr: %s", err)
+		}
+		t.Logf("stderr was: %s", string(stderr))
+
+		stdout, err := os.ReadFile(filepath.Join(tempDir, "stdout"))
+		if err != nil {
+			t.Logf("error reading stdout: %s", err)
+		}
+		t.Logf("stdout was: %s", string(stdout))
+
+		glob := fmt.Sprintf("%s-*.ndjson", filepath.Join(tempDir, beatName))
+		files, err := filepath.Glob(glob)
+		for _, f := range files {
+			contents, err := os.ReadFile(f)
+			if err != nil {
+				t.Logf("error reading %s: %s", f, err)
+			}
+			maxlen := 2048
+			if len(contents) >= maxlen {
+				t.Logf("Last %d bytes of %s:\n%s", maxlen, f, string(contents[len(contents)-maxlen:]))
+			} else {
+				t.Logf("%s\n%s", string(contents), f)
+			}
+		}
+	})
 	return &p
 }
 
@@ -219,6 +251,47 @@ func (b *BeatProc) Stop() {
 	}
 }
 
+// LogMatch tests each line of the logfile to see if contains any
+// match of the provided regular expression.  It will open the log
+// file on every call, read until EOF, then close it.  LogContains
+// will be faster so use that if possible.
+func (b *BeatProc) LogMatch(match string) bool {
+	re := regexp.MustCompile(match)
+	logFile := b.openLogFile()
+	_, err := logFile.Seek(b.logFileOffset, os.SEEK_SET)
+	if err != nil {
+		b.t.Fatalf("could not set offset for '%s': %s", logFile.Name(), err)
+	}
+
+	defer func() {
+		if err := logFile.Close(); err != nil {
+			// That's not quite a test error, but it can impact
+			// next executions of LogContains, so treat it as an error
+			b.t.Errorf("could not close log file: %s", err)
+		}
+	}()
+
+	r := bufio.NewReader(logFile)
+	for {
+		data, err := r.ReadBytes('\n')
+		line := string(data)
+		b.logFileOffset += int64(len(data))
+
+		if err != nil {
+			if err != io.EOF {
+				b.t.Fatalf("error reading log file '%s': %s", logFile.Name(), err)
+			}
+			break
+		}
+
+		if re.MatchString(line) {
+			return true
+		}
+	}
+
+	return false
+}
+
 // LogContains looks for `s` as a substring of every log line,
 // it will open the log file on every call, read it until EOF,
 // then close it.
@@ -242,7 +315,7 @@ func (b *BeatProc) LogContains(s string) bool {
 	for {
 		data, err := r.ReadBytes('\n')
 		line := string(data)
-		b.logFileOffset += int64(len(line))
+		b.logFileOffset += int64(len(data))
 
 		if err != nil {
 			if err != io.EOF {

--- a/libbeat/tests/integration/framework.go
+++ b/libbeat/tests/integration/framework.go
@@ -123,31 +123,27 @@ func NewBeat(t *testing.T, beatName, binary string, args ...string) *BeatProc {
 		if !t.Failed() {
 			return
 		}
-		stderr, err := os.ReadFile(filepath.Join(tempDir, "stderr"))
+		var maxlen int64 = 2048
+		stderr, err := readLastNBytes(filepath.Join(tempDir, "stderr"), maxlen)
 		if err != nil {
 			t.Logf("error reading stderr: %s", err)
 		}
-		t.Logf("stderr was: %s", string(stderr))
+		t.Logf("Last %d bytes of stderr:\n%s", len(stderr), string(stderr))
 
-		stdout, err := os.ReadFile(filepath.Join(tempDir, "stdout"))
+		stdout, err := readLastNBytes(filepath.Join(tempDir, "stdout"), maxlen)
 		if err != nil {
 			t.Logf("error reading stdout: %s", err)
 		}
-		t.Logf("stdout was: %s", string(stdout))
+		t.Logf("Last %d bytes of stdout:\n%s", len(stdout), string(stdout))
 
 		glob := fmt.Sprintf("%s-*.ndjson", filepath.Join(tempDir, beatName))
 		files, err := filepath.Glob(glob)
 		for _, f := range files {
-			contents, err := os.ReadFile(f)
+			contents, err := readLastNBytes(f, maxlen)
 			if err != nil {
 				t.Logf("error reading %s: %s", f, err)
 			}
-			maxlen := 2048
-			if len(contents) >= maxlen {
-				t.Logf("Last %d bytes of %s:\n%s", maxlen, f, string(contents[len(contents)-maxlen:]))
-			} else {
-				t.Logf("%s\n%s", string(contents), f)
-			}
+			t.Logf("Last %d bytes of %s:\n%s", len(contents), f, string(contents))
 		}
 	})
 	return &p
@@ -636,4 +632,26 @@ func FormatDataStreamSearchURL(t *testing.T, srcURL url.URL, dataStream string) 
 	}
 	srcURL.Path = path
 	return srcURL, nil
+}
+
+func readLastNBytes(filename string, numBytes int64) ([]byte, error) {
+	f, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("error opening %s: %w", filename, err)
+	}
+	fInfo, err := f.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("error stating %s: %w", filename, err)
+	}
+	var startPosition int64
+	if fInfo.Size() >= numBytes {
+		startPosition = fInfo.Size() - numBytes
+	} else {
+		startPosition = 0
+	}
+	_, err = f.Seek(startPosition, io.SeekStart)
+	if err != nil {
+		return nil, fmt.Errorf("error seeking to %d in %s: %w", startPosition, filename, err)
+	}
+	return io.ReadAll(f)
 }

--- a/libbeat/tests/integration/template_test.go
+++ b/libbeat/tests/integration/template_test.go
@@ -223,7 +223,9 @@ logging:
 	mockbeat.Start()
 	mockbeat.WaitForLogs("mockbeat start running.", 60*time.Second)
 	mockbeat.WaitForLogs("Template with name \\\"mockbeat-9.9.9\\\" loaded.", 20*time.Second)
-	mockbeat.WaitForLogs("PublishEvents: 1 events have been published", 20*time.Second)
+	require.Eventually(t, func() bool {
+		return mockbeat.LogMatch("PublishEvents: [[:digit:]]+ events have been published")
+	}, 20*time.Second, 100*time.Millisecond, "looking for PublishEvents")
 
 	status, body, err := HttpDo(t, http.MethodGet, indexURL)
 	require.NoError(t, err)
@@ -293,7 +295,9 @@ logging:
 	mockbeat.WriteConfigFile(cfg)
 	mockbeat.Start()
 	mockbeat.WaitForLogs("mockbeat start running.", 60*time.Second)
-	mockbeat.WaitForLogs("PublishEvents: 1 events have been published", 20*time.Second)
+	require.Eventually(t, func() bool {
+		return mockbeat.LogMatch("PublishEvents: [[:digit:]]+ events have been published")
+	}, 20*time.Second, 100*time.Millisecond, "looking for PublishEvents")
 
 	u := fmt.Sprintf("%s/_index_template/%s", esUrl.String(), datastream)
 	r, _ := http.Get(u)

